### PR TITLE
Add support for file name patterns for server registration

### PR DESCRIPTION
**Pull request by gayanper**
## Purpose
In addition to extension, file name pattern support is added. The first
priority is given for extension and then for the file name pattern when
resolving server definitions.

## Goals
To be able to register the language server based on file name pattern like "application.*\.properties" for Spring Boot property file for example.

### CODE REVIEW

File: src/main/java/org/wso2/lsp4intellij/IntellijLanguageClient.java [hunk 1]
```diff
@@ -140,8 +140,24 @@ public static void editorOpened(Editor editor) {
         if (file != null) {
             ApplicationUtils.pool(() -> {
                 String ext = file.getExtension();
+                final String fileName = file.getName();
                 LOG.info("Opened " + file.getName());
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/IntellijLanguageClient.java]:
```suggestion
                LOG.info("Opened " + fileName);
```

File: src/main/java/org/wso2/lsp4intellij/IntellijLanguageClient.java [hunk 2]
```diff
@@ -140,8 +140,24 @@ public static void editorOpened(Editor editor) {
...
+
+                // The ext can be a file name or it can be a file pattern or the extension.
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/IntellijLanguageClient.java]:
```suggestion
                // The ext can either be a file extension or a file pattern(regex expression).
```

**Comment by NipunaRanasinghe** [comment]
Seems there is a merge conflict after merging the Timeouts API :/ Shall we fix?

**Comment by gayanper** [comment]
Sure i will check the change tonight and update the PR

**Comment by gayanper** [comment]
@NipunaRanasinghe conflicts resolved

**Pull request merged**